src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,136 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double MULTIPLICATION = 1.5;
+
+    private T[] arrayList;
+    private int size;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    private int getIndex(T value) {
+        int index = -1;
+        for (int i = 0; i < size; i++) {
+            if (arrayList[i] == value || value != null && value.equals(arrayList[i])) {
+                return i;
+            }
+        }
+        return index;
+    }
+
     @Override
     public void add(T value) {
-
+        resize();
+        arrayList[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+
+        checkIndex(index);
+        resize();
 
+        System.arraycopy(arrayList, index, arrayList, index + 1, size - index);
+        arrayList[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return arrayList[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        arrayList[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+
+        T removed = get(index);
```

Comment:
```suggestion
        T removed = arrayList[index];
```
тут можна одразу дістати